app/assets/javascripts/Components/Result/url_viewer.jsx

```diff
@@ -3,19 +3,82 @@ import React from "react";
 export class URLViewer extends React.Component {
   constructor(props) {
     super(props);
-    //TODO: Remove state url and replace its usage with url props when backend is working
     this.state = {
-      url: "https://www.youtube.com/embed/dQw4w9WgXcQ", // Sample Youtube embed link
-      //url: "https://drive.google.com/file/d/1yQh8bzo4IzmtSt-qZ0Vb-BzzEXsp45v_/preview" // sample google drive embed link
-      //url: "https://play.library.utoronto.ca/watch/4493e554354a7e51bdf90f3197185b5f" // sample mymedia link
+      url: "",
+      show_iframe_preview: false,
     };
   }
 
+  componentDidMount() {
+    try {
+      const url = new URL(this.props.content);
+      const youtube_id_is_set = this.configureYoutubePreview(url.toString());
+      if (!youtube_id_is_set) {
+        switch (url.hostname) {
+          case "docs.google.com":
+          case "drive.google.com":
+            this.configureGoogleDrivePreview(url.toString());
+            break;
+          case "play.library.utoronto.ca":
+            this.setState({
+              url: url.toString(),
+              show_iframe_preview: false,
+            });
+            break;
+          default:
+            this.setDefaultState();
+        }
+      }
+    } catch (e) {
+      this.setDefaultState();
+    }
+  }
+
+  setDefaultState = () => {
+    this.setState({
+      url: "",
+      show_iframe_preview: false,
+    });
+  };
+
+  configureGoogleDrivePreview = url => {
+    const regex = /\/d\/(.+)\//;
```

Comment:
I have checked for a more valid way to do this. The best solution I can find is this https://stackoverflow.com/questions/16840038/easiest-way-to-get-file-id-from-url-on-google-apps-script though I hesitated to use it since it looks even more generic than this. 
The api for google drive only allows getting files by their id or searching for it using various parameters about the file. There is nothing they provide that allows extraction of an id from a url.

Reply:
Ok correct me if I'm wrong but if it's either a drive.google.com or a docs.google.com link as long as the action (last bit of the pathname in the url) is 'preview' it will work right?

So what if we just did that... change the call to `configureGoogleDrivePreview` so that the argument is a url object (not a string) and then we can just do something like:

```js
  configureGoogleDrivePreview = url => {
    url.pathname = url.pathname.replace(/(\/[^\/]+)$/, '/edit')
    this.setState({
      url: url.toString(),
      show_iframe_preview: true,
    });
  }
```

And that way we don't have to worry about extracting ids at all.